src/tree-manager/flatten-tree.js

```diff
@@ -138,6 +140,9 @@ function walkNodes({ nodes, list = new Map(), parent, depth = 0, simple, showPar
     } else {
       node._id = node.id || `${i}`
     }
+    if (depth === 0 && rootPrefixId && rootPrefixId.length > 0) {
```

Comment:
`rootPrefixId` is a string and the beauty of the language is that just the first check would suffice. I would imagine that the length check may be needed for cases where it's all whitespaces (so you'd trim it and then check for length or something like that). 

However, I think we don't need this extra `if`. It can be covered in the `else` branch at L141

```
} else {
      node._id = node.id || `${rootPrefixId ? `${rootPrefixId}-${i}` : i}`
```

The logic is: 
 - use `node.id` if explicitly passed (this is current contract)
 - otherwise, create our own:
   - prefix `rootPrefixId` if passed
   - otherwise, just use the counter